### PR TITLE
refactor: Refactor method findById, checking for exceptions before ma…

### PR DESCRIPTION
**Pull request by MackowiakPiotr**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java [hunk 1]
```diff
@@ -61,10 +61,11 @@ public List<FinancialTransactionDTO> getFinancialTransactionsByWalletId(@Min(1)
 
     @Override
     public FinancialTransactionDTO findById(@Min(1) @NotNull Long id) {
-        Optional<FinancialTransaction> financialTransaction = financialTransactionRepository.findById(id);
-        return financialTransaction.map(financialTransactionModelMapper::mapFinancialTransactionEntityToFinancialTransactionDTO)
+        FinancialTransaction financialTransaction = financialTransactionRepository.findById(id)
                 .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT001,
-                        String.format("Financial transaction with id: %d not found", id)));
+                String.format("Financial transaction with id: %d not found", id)));
+        return financialTransactionModelMapper.mapFinancialTransactionEntityToFinancialTransactionDTO(financialTransaction);
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:67]: And with proper code formatting:
```java
    @Override
    public FinancialTransactionDTO findById(@Min(1) @NotNull Long id) {
        FinancialTransaction financialTransaction = financialTransactionRepository.findById(id)
                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT001,
                String.format("Financial transaction with id: %d not found", id)));

        return financialTransactionModelMapper.mapFinancialTransactionEntityToFinancialTransactionDTO(financialTransaction);
    }
```

or
```java
    @Override
    public FinancialTransactionDTO findById(@Min(1) @NotNull Long id) {
        FinancialTransaction financialTransaction = financialTransactionRepository.findById(id)
                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT001,
                String.format("Financial transaction with id: %d not found", id)));
        return financialTransactionModelMapper.mapFinancialTransactionEntityToFinancialTransactionDTO(financialTransaction);
    }
```

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Pull request merged**